infer/crates/kanalizer-rs/src/error.rs

```diff
@@ -0,0 +1,11 @@
+use thiserror::Error;
+
+#[derive(Error, Debug, Clone, PartialEq)]
+pub enum Error {
+    #[error("入力が空です")]
+    EmptyInput,
+    #[error("無効な文字が含まれています：{chars:?}")]
+    InvalidChars { chars: Vec<char> },
+}
```

Comment:
Rust API的には`#[non_exhaustive]`を付けたさがありますね。kanalizer-pyで使えるようにするにはこう？`#[doc(hidden)] __NonExhausitive`というバリアントを足す形でもよいとは思います。

(ちなみにVOICEVOX COREでは[`Error`はopaqueであり、`ErrorKind`を出す形にしています](https://voicevox.github.io/voicevox_core/apis/rust_api/voicevox_core/struct.Error.html)。)

```suggestion
#[derive(Error, Debug, Clone, PartialEq)]
#[non_exhaustive]
pub enum Error {
    #[error("入力が空です")]
    EmptyInput,
    #[error("無効な文字が含まれています：{chars:?}")]
    InvalidChars { chars: Vec<char> },
}

#[doc(hidden)]
impl Error {
    pub fn __kind(&self) -> __FfiErrorKind {
        let (Self::EmptyInput | Self::InvalidChars { .. }) = self;
        __FfiErrorKind::InvalidInput
    }
}

#[doc(hidden)]
pub enum __FfiErrorKind {
    /// 入力が不正。
    InvalidInput,
}
```